### PR TITLE
Issue #4 As a user, I want to set up a new shopping list so I can start tracking purchased items.

### LINKED ISSUE #4: 4. As a user, I want to set up a new shopping list so I can start tracking purchased items.
## Summary 

A shopping list is a set of items associated with the owner's unique id and the name of the shopping list. Users need to be able to create new shopping lists. To do this, the `Home` view should present them with a form that allows them to enter the name of their list. Additionally, we need to do a few things to create that list:

1. Save the token to `localStorage`
2. Save the list to our database using the `createList` function in `firebase.js` 
3. Show the user the list view
## Acceptance criteria
UI-related tasks:
- [x] The `Home` view displays a form that allows users to enter the name of a shopping list and then create a list with that name.
- [x] The input that accepts the name of the item has a semantic `label` element associated with it
- [x] The user can submit this form with both the mouse and the `Enter` key  
- [x]  When the user submits the form, they see a message indicating that the list either was or was not created and saved to the database.

Data-related tasks:
- [x] Clicking the button uses the `createList` function in `src/api/firebase.js` to create a new shopping list for the user.
- [x] The shopping list path is stored in local storage using the `setListPath` function.
- [x] Once the list has been created and saved, the user is redirected to the `List` view.

## Notes
- The `createList` function takes **three** arguments: the current user's id, the current user's email, and the name of the shopping list.
- The `setListPath` function takes **one** argument: the new list token.
- If you get stuck trying to redirect the user, check out the [useNavigate](https://reactrouter.com/en/6.14.2/hooks/use-navigate) hook from `react-rotuer`.

### PR DESCRIPTION
**Pull request by jmahamed**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description

This code adds a form and allows the user to create a list. When the submit button is clicked, the createList function is triggered. An alert tells the user that the list is created.The following is fetched: the userID, the list name and user email. This information is stored in local storage and allows the user to track their purchased items

## Related Issue

Closes  #4 

## Acceptance Criteria

UI-related tasks:

 The Home view displays a form that allows users to enter the name of a shopping list and then create a list with that name.
 The input that accepts the name of the item has a semantic label element associated with it
 The user can submit this form with both the mouse and the Enter key
 When the user submits the form, they see a message indicating that the list either was or was not created and saved to the database.

Data-related tasks:

 Clicking the button uses the createList function in src/api/firebase.js to create a new shopping list for the user.
 The shopping list path is stored in local storage using the setListPath function.
 Once the list has been created and saved, the user is redirected to the List view.

## Type of Changes

UI and data changes

### Before

Screenshot 2024-02-17 191005

### After

Screenshot 2024-02-17 190948
Screenshot 2024-02-17 191033

## Testing Steps / QA Criteria

- sign in
- enter a list name in the input and enter submit
- check firebase to see if it saved a token under your user ID
- return to the home page and you should see your new list

### CODE REVIEW

File: src/views/Home.jsx [hunk 1]
```diff
@@ -1,18 +1,52 @@
...
+
+	const handleSubmit = async (event) => {
+		event.preventDefault();
+		try {
+			const newList = await createList(userId, userEmail, name);
+			setListPath(newList);
+			setName('');
+			alert('The item has been added.');
+			navigate('/list');
+			localStorage.setItem('tcl-shopping-list-path', newList);
```
Comment (redapy) [inline src/views/Home.jsx]: The `setListPath`function saves the token to local storage, so you don't need to do it again with `localStorage.setItem('tcl-shopping-list-path', newList);`

File: src/views/Home.jsx [hunk 2]
```diff
@@ -1,18 +1,52 @@
...
+		} catch (err) {
+			console.error(err);
+		}
```
Comment (redapy) [inline src/views/Home.jsx]: you need to alert the user if the item was not added, instead of only doing a console log.

**Review by redapy — CHANGES_REQUESTED** [review]
Great work, everything works as expected! There are only a few changes that need to be made

File: src/views/Home.jsx (hunk 1 above)
Comment (ijayhub) [reply src/views/Home.jsx]: okay

File: src/views/Home.jsx (hunk 2 above)
Comment (ijayhub) [reply src/views/Home.jsx]: done

Comment (ijayhub) [reply src/views/Home.jsx]: done

File: src/views/Home.jsx (hunk 1 above)
Comment (ijayhub) [reply src/views/Home.jsx]: corrected

**Review by ijayhub — APPROVED** [review]

**Review by redapy — APPROVED** [review]

**Pull request merged**